chapter09/pxd/ex2.pl

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use feature "say";
+
+open my $fh, '<', $ARGV[0]?$ARGV[0]:'input.txt'
+    or die "Can't open input.txt: $!";
+
+my @patterns;
+
+while (<$fh>) {
+    chomp;
+    my $pattern = eval { qr/$_/ }; 
+    if (defined $pattern) {
+    	push @patterns, $pattern; 
+    }
+
+}
+
+if (scalar  @patterns == 0) {
+	say "Nothing to match to... Provide a file will regular expressions to match to..";
+	exit 0;
```

Comment:
It's usually unnecessary to specify an exit code. It'll default to 0, and if you want to end your script unsuccessfully it's better to use a ``die``. You can revisit it if you find yourself needing to exit your script for a number of failure reasons.